docs/designs/output.md

```diff
@@ -128,4 +185,7 @@ File name           | Description
 .xrefmap.json        | A manifest of `uid` and xref specs as described in [xref](xref.md)
 .errors.log          | A report file that contains build errors and warnings. Each line is a json array: `[{level}, {code}, {message}, {file?}, {line?}, {column?}]`.
 
+## Copy Resource
 
+To improve docs publish performance, there is feature to skip copy resource files to the output folder.
+With those two new configs(`OutputType` and `OutputUrlType`) involved, whether need to copy resource should be determined  by `OutputUrlType`: skip copy resource files when it is `Docs`.
```

Comment:
>OutputUrlType [](start = 127, length = 13)

I would suggest having this as a separate option because even for output type `Html`, we can improve perf by avoid copying images.

Reply:
it is not determined by `OutputType`, but `OutputUrl` type, as the requirements list above, `Static build output xcopy deployable` requires the output to be self-contained.

---
In reply to: [412878126](https://github.com/dotnet/docfx/pull/5815#discussion_r412878126) [](ancestors = 412878126)

Reply:
Keeping it a separate option allows us to create a self contained package for docs that can be re-published later. Maybe changing the name from `copyResources` --> `selfContained` describes the intent better.

Reply:
what does this requirements come from?

---
In reply to: [413451101](https://github.com/dotnet/docfx/pull/5815#discussion_r413451101) [](ancestors = 413451101)

Reply:
and in the current implementation, the `CopyResource` config doesn't work when in legacy mode..

---
In reply to: [413454336](https://github.com/dotnet/docfx/pull/5815#discussion_r413454336) [](ancestors = 413454336,413451101)